File: python/meteroid/models/batch_job_detail_response.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .batch_job_id import BatchJobId
from .batch_job_status import BatchJobStatus
from .batch_job_type import BatchJobType


@dataclasses.dataclass
class BatchJobDetailResponse(BaseModel):
    created_at: datetime

    created_by: str

    failed_items: int

    failure_count: int

    has_error_csv: bool

    has_output: bool

    id: BatchJobId

    job_type: BatchJobType

    processed_items: int

    status: BatchJobStatus

    completed_at: t.Optional[datetime] = None

    error_csv_url: t.Optional[str] = None

    input_file_name: t.Optional[str] = None

    input_file_url: t.Optional[str] = None

    output_url: t.Optional[str] = None

    total_items: t.Optional[int] = None

File: go/batch_job_detail_response.go
// this file is @generated
package meteroid

import "time"

type BatchJobDetailResponse struct {
	// RFC 3339 timestamp.
	CompletedAt *time.Time `json:"completed_at,omitempty"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	CreatedBy string `json:"created_by"`

	ErrorCsvUrl *string `json:"error_csv_url,omitempty"`

	FailedItems int32 `json:"failed_items"`

	FailureCount int64 `json:"failure_count"`

	HasErrorCsv bool `json:"has_error_csv"`

	HasOutput bool `json:"has_output"`

	Id BatchJobId `json:"id"`

	InputFileName *string `json:"input_file_name,omitempty"`

	InputFileUrl *string `json:"input_file_url,omitempty"`

	JobType BatchJobType `json:"job_type"`

	OutputUrl *string `json:"output_url,omitempty"`

	ProcessedItems int32 `json:"processed_items"`

	Status BatchJobStatus `json:"status"`

	TotalItems *int32 `json:"total_items,omitempty"`
}
